nemo/collections/cv/modules/non_trainables/non_linearity.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# =============================================================================
+# Copyright (c) 2020 NVIDIA. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+
+import torch
+
+from nemo.backends.pytorch.nm import NonTrainableNM
+from nemo.core.neural_types import AxisKind, AxisType, LogprobsType, NeuralType, VoidType
+from nemo.utils import logging
+from nemo.utils.decorators import add_port_docs
+
+__all__ = ['NonLinearity']
+
+
+class NonLinearity(NonTrainableNM):
+    """
+    Class responsible for applying additional non-linearity along the last axis of the input tensor.
+
+    """
+
+    def __init__(self, type="logsoftmax", sizes=[-1], name=None):
+        """
+        Initializes the object.
+
+        """
+        # Call constructor of parent classes.
+        NonTrainableNM.__init__(self, name=name)
+
+        # Store params.
+        self._type = type
+        self._sizes = sizes
+
+        # Apply the non-linearity along the last dimension.
+        # TODO: if self._type != "logsoftmax"
+        assert type == "logsoftmax"
```

Comment:
please don't use asserts

Reply:
This module implements just the simplest non-linearity that I needed...

Reply:
ok, used get_value_from_dict